Flaky tests in ParseDistSQLHandlerTest due to nondeterministic json element ordering

Two tests in the test class ParseDistSQLHandlerTest, using the [NonDex](https://github.com/TestingResearchIllinois/NonDex) tool:
```
org.apache.shardingsphere.proxy.backend.handler.distsql.ral.advance.ParseDistSQLHandlerTest#assertGetRowDataForMySQL
org.apache.shardingsphere.proxy.backend.handler.distsql.ral.advance.ParseDistSQLHandlerTest#assertGetRowDataForPostgreSQL
```

Command to reproduce using Nondex: 
`mvn-pl proxy/backend edu.illinois:nondex-maven-plugin:2.1.1:nondex -Dtest=org.apache.shardingsphere.proxy.backend.handler.distsql.ral.advance.ParseDistSQLHandlerTest`

The following assertions failed:
```
assertThat(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getLast(), is(new Gson().toJson(statement)));
assertThat(JsonParser.parseString(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getLast().toString()), is(JsonParser.parseString(new Gson().toJson(statement))));
```
The tests fail due to non-deterministic ordering of the Json objects that are being compared for equality.

Error logs with Nondex:
```
[ERROR]   ParseDistSQLHandlerTest.assertGetRowDataForPostgreSQL:95 
Expected: is "{\"from\":{\"tableName\":{\"identifier\":{\"quoteCharacter\":\"NONE\",\"value\":\"t_order\"},\"stopIndex\":20,\"startIndex\":14}},\"projections\":{\"distinctRow\":false,\"projections\":[{\"stopIndex\":7,\"startIndex\":7}],\"startIndex\":7,\"stopIndex\":7},\"parameterMarkerSegments\":[],\"parameterCount\":0,\"commentSegments\":[]}"
     but: was "{\"from\":{\"tableName\":{\"startIndex\":14,\"stopIndex\":20,\"identifier\":{\"value\":\"t_order\",\"quoteCharacter\":\"NONE\"}}},\"projections\":{\"stopIndex\":7,\"distinctRow\":false,\"projections\":[{\"startIndex\":7,\"stopIndex\":7}],\"startIndex\":7},\"commentSegments\":[],\"parameterMarkerSegments\":[],\"parameterCount\":0}"
```

--- a/proxy/backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/advance/ParseDistSQLHandlerTest.java
+++ b/proxy/backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/advance/ParseDistSQLHandlerTest.java
@@ -18,6 +18,7 @@
 package org.apache.shardingsphere.proxy.backend.handler.distsql.ral.advance;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonParser;
 import org.apache.shardingsphere.distsql.parser.statement.rul.sql.ParseStatement;
 import org.apache.shardingsphere.infra.database.type.dialect.MySQLDatabaseType;
 import org.apache.shardingsphere.infra.database.type.dialect.PostgreSQLDatabaseType;
@@ -76,7 +77,7 @@ public final class ParseDistSQLHandlerTest extends ProxyContextRestorer {
         parseDistSQLHandler.next();
         SQLStatement statement = sqlParserRule.getSQLParserEngine("MySQL").parse(sql, false);
         assertThat(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getFirst(), is("MySQLSelectStatement"));
-        assertThat(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getLast(), is(new Gson().toJson(statement)));
+        assertThat(JsonParser.parseString(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getLast().toString()), is(JsonParser.parseString(new Gson().toJson(statement))));
     }
     
     @Test
@@ -89,8 +90,7 @@ public final class ParseDistSQLHandlerTest extends ProxyContextRestorer {
         parseDistSQLHandler.execute();
         parseDistSQLHandler.next();
         SQLStatement statement = sqlParserRule.getSQLParserEngine("PostgreSQL").parse(sql, false);
-        assertThat(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getFirst(), is("PostgreSQLSelectStatement"));
-        assertThat(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getLast(), is(new Gson().toJson(statement)));
+        assertThat(JsonParser.parseString(new LinkedList<>(parseDistSQLHandler.getRowData().getData()).getLast().toString()), is(JsonParser.parseString(new Gson().toJson(statement))));
     }
     
     @Test(expected = SQLParsingException.class)

